src/Avalonia.Base/AvaloniaPropertyRegistry.cs

```diff
@@ -148,6 +150,16 @@ public AvaloniaProperty FindRegistered(AvaloniaObject o, string name)
             return FindRegistered(o.GetType(), name);
         }
 
+        /// <summary>
+        /// Finds a registered property by Id.
+        /// </summary>
+        /// <param name="id">The property Id.</param>
+        /// <returns>The registered property or null if no matching property found.</returns>
+        public AvaloniaProperty FindRegistered(int id)
```

Comment:
The `AvaloniaProperty.Id` property is `internal` as it's meant to be an implementation detail, so this method should probably be internal too.